index.bs

```diff
@@ -2169,16 +2313,65 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
 
 1. Let |document| be |context|'s [=active document=].
 
+1. Let |viewport| be |document|'s [=visual viewport=].
+
 1. Immediately after the next invocation of the [=run the animation frame
    callbacks=] algorithm for |document|:
 
    Issue: This ought to be integrated into the update rendering algorithm
    in some more explicit way.
 
- 1. Let |root rect| be |document|’s [=document element=]'s [=rectangle=].
+1. Let |viewport rect| be a {{DOMRectReadOnly}} with [=x coordinate=] 0, [=y
```

Comment:
This is only used in the otherwise section below in line 2367 so we potentiall can get this moved there.

Reply:
I generally rearranged this section to make it easier to follow.